### PR TITLE
python311Packages.openstacksdk: 3.3.0 -> 4.0.0

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by anthonyroussel** [comment]
Result of `nixpkgs-review pr 338474` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>bashate</li>
    <li>bashate.dist (bashate.dist.dist)</li>
    <li>python311Packages.bindep</li>
    <li>python311Packages.bindep.dist (python311Packages.bindep.dist.dist)</li>
    <li>python311Packages.openstackdocstheme</li>
    <li>python311Packages.openstackdocstheme.dist (python311Packages.openstackdocstheme.dist.dist)</li>
    <li>python311Packages.oslo-config</li>
    <li>python311Packages.oslo-config.dist (python311Packages.oslo-config.dist.dist)</li>
    <li>python311Packages.pbr</li>
    <li>python311Packages.pbr.dist (python311Packages.pbr.dist.dist)</li>
    <li>python311Packages.python-cinderclient</li>
    <li>python311Packages.python-cinderclient.dist (python311Packages.python-cinderclient.dist.dist)</li>
    <li>python311Packages.python-keystoneclient</li>
    <li>python311Packages.python-keystoneclient.dist (python311Packages.python-keystoneclient.dist.dist)</li>
    <li>python311Packages.python-manilaclient</li>
    <li>python311Packages.python-manilaclient.dist (python311Packages.python-manilaclient.dist.dist)</li>
    <li>python311Packages.python-novaclient</li>
    <li>python311Packages.python-novaclient.dist (python311Packages.python-novaclient.dist.dist)</li>
    <li>python311Packages.stevedore</li>
    <li>python311Packages.stevedore.dist (python311Packages.stevedore.dist.dist)</li>
    <li>reno</li>
    <li>reno.dist (reno.dist.dist)</li>
  </ul>
</details>

**Comment by anthonyroussel** [comment]
@ofborg build bashate python311Packages.python-cinderclient reno python311Packages.bindep python311Packages.stevedore python311Packages.python-novaclient python311Packages.oslo-config python311Packages.python-novaclient python311Packages.python-manilaclient python311Packages.python-keystoneclient python311Packages.python-cinderclient python311Packages.pbr python311Packages.openstackdocstheme

**Comment by SuperSandro2000** [comment]
That needs to target staging. Please see the contributing docs on how to do this.

**Comment by anthonyroussel** [comment]
Moved pbr upgrade to https://github.com/NixOS/nixpkgs/pull/338653 to avoid too many rebuilds on this PR

File: pkgs/development/python-modules/bindep/default.nix [hunk 1]
```diff
@@ -1,42 +1,49 @@
...
+  doCheck = false;
+
+  pythonImportsCheck = [ "bindep" ];
 
   meta = with lib; {
     description = "Bindep is a tool for checking the presence of binary packages needed to use an application / library";
     homepage = "https://docs.opendev.org/opendev/bindep/latest/";
     license = licenses.asl20;
-    maintainers = with maintainers; [ melkor333 ];
+    maintainers = teams.openstack.members ++ (with maintainers; [ melkor333 ]);
```
Comment (Melkor333) [inline pkgs/development/python-modules/bindep/default.nix:47]: Could you remove me as maintainer? I only packaged it as a dependency for another tool I needed (IIRC for `ansible-builder/ansible-navigator`). Therefore I'm completely unfamiliar with the tool. :)

Comment (anthonyroussel) [reply pkgs/development/python-modules/bindep/default.nix]: Hello @Melkor333,
No problem, I removed you from the maintainers of this package.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 338474` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.subliminal</li>
    <li>python311Packages.subliminal.dist</li>
    <li>python311Packages.subunit2sql</li>
    <li>python311Packages.subunit2sql.dist</li>
    <li>python312Packages.subliminal</li>
    <li>python312Packages.subliminal.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>ansible-navigator</li>
    <li>ansible-navigator.dist</li>
    <li>python311Packages.e3-core</li>
    <li>python311Packages.e3-core.dist</li>
    <li>python311Packages.e3-testsuite</li>
    <li>python311Packages.e3-testsuite.dist</li>
    <li>python311Packages.jenkins-job-builder</li>
    <li>python311Packages.jenkins-job-builder.dist</li>
    <li>python312Packages.e3-core</li>
    <li>python312Packages.e3-core.dist</li>
    <li>python312Packages.e3-testsuite</li>
    <li>python312Packages.e3-testsuite.dist</li>
  </ul>
</details>
<details>
  <summary>130 packages built:</summary>
  <ul>
    <li>ansible-builder (python312Packages.ansible-builder)</li>
    <li>ansible-builder.dist (python312Packages.ansible-builder.dist)</li>
    <li>bandit (python312Packages.bandit)</li>
    <li>bandit.dist (python312Packages.bandit.dist)</li>
    <li>barbicanclient (python311Packages.python-barbicanclient)</li>
    <li>barbicanclient.dist (python311Packages.python-barbicanclient.dist)</li>
    <li>bashate</li>
    <li>bashate.dist</li>
    <li>glanceclient (python311Packages.python-glanceclient)</li>
    <li>glanceclient.dist (python311Packages.python-glanceclient.dist)</li>
    <li>heatclient (python311Packages.python-heatclient)</li>
    <li>heatclient.dist (python311Packages.python-heatclient.dist)</li>
    <li>home-assistant-component-tests.nx584</li>
    <li>ironicclient (python311Packages.python-ironicclient)</li>
    <li>ironicclient.dist (python311Packages.python-ironicclient.dist)</li>
    <li>kolla</li>
    <li>kolla.dist</li>
    <li>localstack (python312Packages.localstack)</li>
    <li>localstack.dist (python312Packages.localstack.dist)</li>
    <li>manilaclient (python311Packages.python-manilaclient)</li>
    <li>manilaclient.dist (python311Packages.python-manilaclient.dist)</li>
    <li>nox</li>
    <li>nox.dist</li>
    <li>openstackclient (python311Packages.python-openstackclient)</li>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>openstackclient.dist (python311Packages.python-openstackclient.dist)</li>
    <li>papis (python312Packages.papis)</li>
    <li>papis.dist (python312Packages.papis.dist)</li>
    <li>python311Packages.bandit</li>
    <li>python311Packages.bandit.dist</li>
    <li>python311Packages.bindep</li>
    <li>python311Packages.bindep.dist</li>
    <li>python311Packages.bugwarrior</li>
    <li>python311Packages.bugwarrior.dist</li>
    <li>python311Packages.cliff</li>
    <li>python311Packages.cliff.dist</li>
    <li>python311Packages.debtcollector</li>
    <li>python311Packages.debtcollector.dist</li>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.dogpile-cache</li>
    <li>python311Packages.dogpile-cache.dist</li>
    <li>python311Packages.hacking</li>
    <li>python311Packages.hacking.dist</li>
    <li>python311Packages.keystoneauth1</li>
    <li>python311Packages.keystoneauth1.dist</li>
    <li>python311Packages.localstack</li>
    <li>python311Packages.localstack-ext</li>
    <li>python311Packages.localstack-ext.dist</li>
    <li>python311Packages.localstack.dist</li>
    <li>python311Packages.openstackdocstheme</li>
    <li>python311Packages.openstackdocstheme.dist</li>
    <li>python311Packages.openstacksdk</li>
    <li>python311Packages.openstacksdk.dist</li>
    <li>python311Packages.openstacksdk.man</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osc-lib</li>
    <li>python311Packages.osc-lib.dist</li>
    <li>python311Packages.oslo-concurrency</li>
    <li>python311Packages.oslo-concurrency.dist</li>
    <li>python311Packages.oslo-config</li>
    <li>python311Packages.oslo-config.dist</li>
    <li>python311Packages.oslo-context</li>
    <li>python311Packages.oslo-context.dist</li>
    <li>python311Packages.oslo-db</li>
    <li>python311Packages.oslo-db.dist</li>
    <li>python311Packages.oslo-i18n</li>
    <li>python311Packages.oslo-i18n.dist</li>
    <li>python311Packages.oslo-log</li>
    <li>python311Packages.oslo-log.dist</li>
    <li>python311Packages.oslo-serialization</li>
    <li>python311Packages.oslo-serialization.dist</li>
    <li>python311Packages.oslo-utils</li>
    <li>python311Packages.oslo-utils.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.papis</li>
    <li>python311Packages.papis.dist</li>
    <li>python311Packages.plux</li>
    <li>python311Packages.plux.dist</li>
    <li>python311Packages.pynx584</li>
    <li>python311Packages.pynx584.dist</li>
    <li>python311Packages.python-cinderclient</li>
    <li>python311Packages.python-cinderclient.dist</li>
    <li>python311Packages.python-designateclient</li>
    <li>python311Packages.python-designateclient.dist</li>
    <li>python311Packages.python-jenkins</li>
    <li>python311Packages.python-jenkins.dist</li>
    <li>python311Packages.python-keystoneclient</li>
    <li>python311Packages.python-keystoneclient.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-novaclient</li>
    <li>python311Packages.python-novaclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>swiftclient (python311Packages.python-swiftclient)</li>
    <li>swiftclient.dist (python311Packages.python-swiftclient.dist)</li>
    <li>python311Packages.stestr</li>
    <li>python311Packages.stestr.dist</li>
    <li>python311Packages.stevedore</li>
    <li>python311Packages.stevedore.dist</li>
    <li>python311Packages.tempest</li>
    <li>python311Packages.tempest.dist</li>
    <li>python311Packages.tern</li>
    <li>python311Packages.tern.dist</li>
    <li>python311Packages.virtualenvwrapper</li>
    <li>python311Packages.virtualenvwrapper.dist</li>
    <li>python312Packages.bindep</li>
    <li>python312Packages.bindep.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
    <li>python312Packages.dogpile-cache</li>
    <li>python312Packages.dogpile-cache.dist</li>
    <li>python312Packages.localstack-ext</li>
    <li>python312Packages.localstack-ext.dist</li>
    <li>python312Packages.plux</li>
    <li>python312Packages.plux.dist</li>
    <li>python312Packages.pynx584</li>
    <li>python312Packages.pynx584.dist</li>
    <li>python312Packages.stevedore</li>
    <li>python312Packages.stevedore.dist</li>
    <li>python312Packages.tern</li>
    <li>python312Packages.tern.dist</li>
    <li>python312Packages.virtualenvwrapper</li>
    <li>python312Packages.virtualenvwrapper.dist</li>
    <li>reno</li>
    <li>reno.dist</li>
  </ul>
</details>

**Comment by anthonyroussel** [comment]
Result of `nixpkgs-review pr 338474` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.subliminal</li>
    <li>python311Packages.subliminal.dist</li>
    <li>python311Packages.subunit2sql</li>
    <li>python311Packages.subunit2sql.dist</li>
    <li>python312Packages.subliminal</li>
    <li>python312Packages.subliminal.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>python311Packages.e3-core</li>
    <li>python311Packages.e3-core.dist</li>
    <li>python311Packages.e3-testsuite</li>
    <li>python311Packages.e3-testsuite.dist</li>
    <li>python311Packages.jenkins-job-builder</li>
    <li>python311Packages.jenkins-job-builder.dist</li>
    <li>python312Packages.e3-core</li>
    <li>python312Packages.e3-core.dist</li>
    <li>python312Packages.e3-testsuite</li>
    <li>python312Packages.e3-testsuite.dist</li>
  </ul>
</details>
<details>
  <summary>140 packages built:</summary>
  <ul>
    <li>ansible-builder (python312Packages.ansible-builder)</li>
    <li>ansible-builder.dist (python312Packages.ansible-builder.dist)</li>
    <li>ansible-navigator</li>
    <li>ansible-navigator.dist</li>
    <li>bandit (python312Packages.bandit)</li>
    <li>bandit.dist (python312Packages.bandit.dist)</li>
    <li>barbicanclient (python311Packages.python-barbicanclient)</li>
    <li>barbicanclient.dist (python311Packages.python-barbicanclient.dist)</li>
    <li>bashate</li>
    <li>bashate.dist</li>
    <li>glanceclient (python311Packages.python-glanceclient)</li>
    <li>glanceclient.dist (python311Packages.python-glanceclient.dist)</li>
    <li>heatclient (python311Packages.python-heatclient)</li>
    <li>heatclient.dist (python311Packages.python-heatclient.dist)</li>
    <li>home-assistant-component-tests.nx584</li>
    <li>ironicclient (python311Packages.python-ironicclient)</li>
    <li>ironicclient.dist (python311Packages.python-ironicclient.dist)</li>
    <li>kolla</li>
    <li>kolla.dist</li>
    <li>localstack (python312Packages.localstack)</li>
    <li>localstack.dist (python312Packages.localstack.dist)</li>
    <li>magnumclient (python311Packages.python-magnumclient)</li>
    <li>magnumclient.dist (python311Packages.python-magnumclient.dist)</li>
    <li>manilaclient (python311Packages.python-manilaclient)</li>
    <li>manilaclient.dist (python311Packages.python-manilaclient.dist)</li>
    <li>mistralclient (python311Packages.python-mistralclient)</li>
    <li>mistralclient.dist (python311Packages.python-mistralclient.dist)</li>
    <li>nox</li>
    <li>nox.dist</li>
    <li>openstackclient (python311Packages.python-openstackclient)</li>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>openstackclient.dist (python311Packages.python-openstackclient.dist)</li>
    <li>papis (python312Packages.papis)</li>
    <li>papis.dist (python312Packages.papis.dist)</li>
    <li>python311Packages.ansible-builder</li>
    <li>python311Packages.ansible-builder.dist</li>
    <li>python311Packages.bandit</li>
    <li>python311Packages.bandit.dist</li>
    <li>python311Packages.bindep</li>
    <li>python311Packages.bindep.dist</li>
    <li>python311Packages.bugwarrior</li>
    <li>python311Packages.bugwarrior.dist</li>
    <li>python311Packages.cliff</li>
    <li>python311Packages.cliff.dist</li>
    <li>python311Packages.debtcollector</li>
    <li>python311Packages.debtcollector.dist</li>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.dogpile-cache</li>
    <li>python311Packages.dogpile-cache.dist</li>
    <li>python311Packages.hacking</li>
    <li>python311Packages.hacking.dist</li>
    <li>python311Packages.keystoneauth1</li>
    <li>python311Packages.keystoneauth1.dist</li>
    <li>python311Packages.localstack</li>
    <li>python311Packages.localstack-ext</li>
    <li>python311Packages.localstack-ext.dist</li>
    <li>python311Packages.localstack.dist</li>
    <li>python311Packages.openstackdocstheme</li>
    <li>python311Packages.openstackdocstheme.dist</li>
    <li>python311Packages.openstacksdk</li>
    <li>python311Packages.openstacksdk.dist</li>
    <li>python311Packages.openstacksdk.man</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osc-lib</li>
    <li>python311Packages.osc-lib.dist</li>
    <li>python311Packages.oslo-concurrency</li>
    <li>python311Packages.oslo-concurrency.dist</li>
    <li>python311Packages.oslo-config</li>
    <li>python311Packages.oslo-config.dist</li>
    <li>python311Packages.oslo-context</li>
    <li>python311Packages.oslo-context.dist</li>
    <li>python311Packages.oslo-db</li>
    <li>python311Packages.oslo-db.dist</li>
    <li>python311Packages.oslo-i18n</li>
    <li>python311Packages.oslo-i18n.dist</li>
    <li>python311Packages.oslo-log</li>
    <li>python311Packages.oslo-log.dist</li>
    <li>python311Packages.oslo-serialization</li>
    <li>python311Packages.oslo-serialization.dist</li>
    <li>python311Packages.oslo-utils</li>
    <li>python311Packages.oslo-utils.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.papis</li>
    <li>python311Packages.papis.dist</li>
    <li>python311Packages.plux</li>
    <li>python311Packages.plux.dist</li>
    <li>python311Packages.pynx584</li>
    <li>python311Packages.pynx584.dist</li>
    <li>python311Packages.python-cinderclient</li>
    <li>python311Packages.python-cinderclient.dist</li>
    <li>python311Packages.python-designateclient</li>
    <li>python311Packages.python-designateclient.dist</li>
    <li>python311Packages.python-jenkins</li>
    <li>python311Packages.python-jenkins.dist</li>
    <li>python311Packages.python-keystoneclient</li>
    <li>python311Packages.python-keystoneclient.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-novaclient</li>
    <li>python311Packages.python-novaclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>swiftclient (python311Packages.python-swiftclient)</li>
    <li>swiftclient.dist (python311Packages.python-swiftclient.dist)</li>
    <li>troveclient (python311Packages.python-troveclient)</li>
    <li>troveclient.dist (python311Packages.python-troveclient.dist)</li>
    <li>python311Packages.stestr</li>
    <li>python311Packages.stestr.dist</li>
    <li>python311Packages.stevedore</li>
    <li>python311Packages.stevedore.dist</li>
    <li>python311Packages.tempest</li>
    <li>python311Packages.tempest.dist</li>
    <li>python311Packages.tern</li>
    <li>python311Packages.tern.dist</li>
    <li>python311Packages.virtualenvwrapper</li>
    <li>python311Packages.virtualenvwrapper.dist</li>
    <li>python312Packages.bindep</li>
    <li>python312Packages.bindep.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
    <li>python312Packages.dogpile-cache</li>
    <li>python312Packages.dogpile-cache.dist</li>
    <li>python312Packages.localstack-ext</li>
    <li>python312Packages.localstack-ext.dist</li>
    <li>python312Packages.plux</li>
    <li>python312Packages.plux.dist</li>
    <li>python312Packages.pynx584</li>
    <li>python312Packages.pynx584.dist</li>
    <li>python312Packages.stevedore</li>
    <li>python312Packages.stevedore.dist</li>
    <li>python312Packages.tern</li>
    <li>python312Packages.tern.dist</li>
    <li>python312Packages.virtualenvwrapper</li>
    <li>python312Packages.virtualenvwrapper.dist</li>
    <li>reno</li>
    <li>reno.dist</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> * python311Packages.e3-core

Fails to build because of a bad patch in itself.

```
Running phase: unpackPhase
@nix {"action":"setPhase","phase":"unpackPhase"}
unpacking source archive /nix/store/c66dcj38fhsg4phv9y4i4m39gh9l8kjd-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
Running phase: patchPhase
@nix {"action":"setPhase","phase":"patchPhase"}
applying patch /nix/store/dx91xg3s001wzgi0z09iqv18nbq7r6b3-0001-use-distro-over-ld.patch
can't find file to patch at input line 18
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|From 189681bbfb703a7026ca7bbb3b21ef554807b144 Mon Sep 17 00:00:00 2001
|From: tali auster <taliauster@gmail.com>
|Date: Wed, 15 Nov 2023 12:15:34 -0700
|Subject: [PATCH] use distro over ld
|
|The `ld` module (linux distribution) was renamed to `distro`, presumably
|so as not to subsume binutils name.
|
|---
| setup.py              | 2 +-
| src/e3/os/platform.py | 2 +-
| 2 files changed, 2 insertions(+), 2 deletions(-)
|
|diff --git a/setup.py b/setup.py
|index c32f46f..de1ada6 100644
|--- a/setup.py
|+++ b/setup.py
--------------------------
File to patch:
Skip this patch? [y]
Skipping patch.
1 out of 1 hunk ignored
patching file src/e3/os/platform.py
Hunk #1 succeeded at 76 (offset -2 lines).
```

**Pull request merged**